borg/hashindex.pyx

```diff
@@ -63,7 +77,9 @@ cdef class IndexBase:
             path = os.fsencode(path)
             self.index = hashindex_read(path)
             if not self.index:
-                raise Exception('hashindex_read failed')
+                if errno:
+                    raise OSError(errno, decoded_strerror(errno), os.fsdecode(path))
```

Comment:
fsdecode uses surrogateescape (on *nix); shouldn't explode. strerror on the other hand has to be in lc_messages locale....